annotation/tests/test_annotations.py

```diff
@@ -482,3 +590,571 @@ def test_construct_annotated_doc_db_error(
+    ) as mock_debug:
+        load_validated_pages_for_particular_rev(
+            annotated_doc, page_revision_list_all[1][0], TEST_TENANT, []
+        )
+        mock_debug.assert_called_once_with(
+            "load_validated_pages_for_particular_rev"
+        )
+        mock_convert_prefix.assert_called_once_with(TEST_TENANT)
+        mock_load_page.assert_called_once_with(
+            None,
+            [],
+            TEST_TENANT,
+            1,
+            annotated_doc.user,
+            page_revision_list_all[1][0],
+            False,
+        )
+    assert page_revision_list_all[1][0]["page_id"] is None
+
+
+def test_construct_particular_rev_response(annotated_doc: AnnotatedDoc):
+    page = {
+        "page_num": 1,
+        "size": {"width": 6.0, "height": 9.0},
+        "objs": [],
+    }
+
+    def mock_loaded_pages_append(
+        revision, page_revision, tenant, loaded_pages
+    ):
+        loaded_pages.append({**page})
+
+    with patch(
+        "annotation.annotations.main.load_annotated_pages_for_particular_rev",
+        side_effect=mock_loaded_pages_append,
+    ) as mock_load_annotated, patch(
+        "annotation.annotations.main.load_validated_pages_for_particular_rev",
+        side_effect=mock_loaded_pages_append,
+    ) as mock_load_validated, patch(
+        "annotation.annotations.main.logger_.debug"
+    ) as mock_debug:
+        result = construct_particular_rev_response(annotated_doc)
+        mock_debug.assert_has_calls(
+            [call("Loaded %s revisions", 2), call("Building response")]
+        )
+        mock_load_annotated.assert_called_once_with(
+            annotated_doc,
+            {"job_id": annotated_doc.job_id, "file_id": annotated_doc.file_id},
+            annotated_doc.tenant,
+            [{**page}, {**page}],
+        )
+        mock_load_validated.assert_called_once_with(
+            annotated_doc,
+            {"job_id": annotated_doc.job_id, "file_id": annotated_doc.file_id},
+            annotated_doc.tenant,
+            [{**page}, {**page}],
+        )
+
+        expected_result = ParticularRevisionSchema(
+            revision=annotated_doc.revision,
+            user=annotated_doc.user,
+            pipeline=annotated_doc.pipeline,
+            date=annotated_doc.date,
+            pages=[{**page}, {**page}],
+            validated=annotated_doc.validated,
+            failed_validation_pages=annotated_doc.failed_validation_pages,
+            categories=annotated_doc.categories,
+            similar_revisions=None,
+            links_json=annotated_doc.links_json,
+        )
+
+        assert result == expected_result
+
+
+def test_construct_particular_rev_response_error(annotated_doc: AnnotatedDoc):
+    with patch(
+        "annotation.annotations.main.load_annotated_pages_for_particular_rev",
+        side_effect=Exception,
+    ), patch(
+        "annotation.annotations.main.logger_.exception"
+    ) as mock_logger, pytest.raises(
```

Comment:
Maybe you could extract pytest.raises into a separate with statement nested within the surrounding context for better readability.